File: mobile/lib/domain/models/pin_update_request.dart
/// doc/API_SPEC.md 6.4 — PinUpdateRequest
class PinUpdateRequest {
  const PinUpdateRequest({
    this.latitude,
    this.longitude,
    this.description,
  });

  final double? latitude;
  final double? longitude;
  final String? description;

  Map<String, dynamic> toJson() => {
        if (latitude != null) 'latitude': latitude,
        if (longitude != null) 'longitude': longitude,
        if (description != null) 'description': description,
      };
}

File: src/main/java/com/example/sns/dto/request/PinUpdateRequest.java
package com.example.sns.dto.request;

import jakarta.validation.constraints.Size;

/**
 * Pin 수정 요청.
 * 프론트엔드 호환: title, category 필드 추가.
 */
public record PinUpdateRequest(
        @Size(max = 200)
        String title,

        @Size(max = 500)
        String description,

        String category,

        Double latitude,
        Double longitude
) {
}
